### ISSUE TITLE
undefined method `line' for nil:NilClass

### ISSUE THREAD

**Issue by decodedigital** [brigade/scss-lint#446]
Trying to run SCSS Lint on a file and the following mixin is causing an error:

```scss
@mixin font-size($keyword) {
    $size: map-get($font-sizes, $keyword);

    @if $size == null {
        @warn 'Font size ‘#{$keyword}’ does not exist';
    } @else {
        @include rem(font-size, $size);
    }
}
```

When I remove the `@else { ... }` clause from the mixin the error disappears.

Here is the full output from terminal. I'm running OS X Mavericks 10.9.4 and this is a brand new install of SCSS Lint (version 0.37.0).

```
SCSSLint::Linter::Indentation raised unexpected error linting file base/_type.scss: 'undefined method `line' for nil:NilClass'
/Library/Ruby/Gems/2.0.0/gems/scss-lint-0.37.0/lib/scss_lint/linter/indentation.rb:194:in `node_indent'
/Library/Ruby/Gems/2.0.0/gems/scss-lint-0.37.0/lib/scss_lint/linter/indentation.rb:184:in `one_shift_greater_than_parent?'
/Library/Ruby/Gems/2.0.0/gems/scss-lint-0.37.0/lib/scss_lint/linter/indentation.rb:159:in `check_arbitrary_indent'
/Library/Ruby/Gems/2.0.0/gems/scss-lint-0.37.0/lib/scss_lint/linter/indentation.rb:53:in `check_indent_width'
/Library/Ruby/Gems/2.0.0/gems/scss-lint-0.37.0/lib/scss_lint/linter/indentation.rb:39:in `check_indentation'
/Library/Ruby/Gems/2.0.0/gems/scss-lint-0.37.0/lib/scss_lint/linter/indentation.rb:17:in `check_and_visit_children'
/Library/Ruby/Gems/2.0.0/gems/scss-lint-0.37.0/lib/scss_lint/linter/indentation.rb:61:in `visit_if'
/Library/Ruby/Gems/2.0.0/gems/sass-3.4.13/lib/sass/tree/visitors/base.rb:36:in `visit'
/Library/Ruby/Gems/2.0.0/gems/scss-lint-0.37.0/lib/scss_lint/linter.rb:127:in `visit'
/Library/Ruby/Gems/2.0.0/gems/scss-lint-0.37.0/lib/scss_lint/linter/indentation.rb:62:in `visit_if'
/Library/Ruby/Gems/2.0.0/gems/sass-3.4.13/lib/sass/tree/visitors/base.rb:36:in `visit'
/Library/Ruby/Gems/2.0.0/gems/scss-lint-0.37.0/lib/scss_lint/linter.rb:127:in `visit'
/Library/Ruby/Gems/2.0.0/gems/scss-lint-0.37.0/lib/scss_lint/linter.rb:138:in `block in visit_children'
/Library/Ruby/Gems/2.0.0/gems/scss-lint-0.37.0/lib/scss_lint/linter.rb:136:in `each'
/Library/Ruby/Gems/2.0.0/gems/scss-lint-0.37.0/lib/scss_lint/linter.rb:136:in `visit_children'
/Library/Ruby/Gems/2.0.0/gems/sass-3.4.13/lib/sass/tree/visitors/base.rb:36:in `block in visit'
/Library/Ruby/Gems/2.0.0/gems/scss-lint-0.37.0/lib/scss_lint/linter/indentation.rb:20:in `check_and_visit_children'
/Library/Ruby/Gems/2.0.0/gems/sass-3.4.13/lib/sass/tree/visitors/base.rb:36:in `visit'
/Library/Ruby/Gems/2.0.0/gems/scss-lint-0.37.0/lib/scss_lint/linter.rb:127:in `visit'
/Library/Ruby/Gems/2.0.0/gems/scss-lint-0.37.0/lib/scss_lint/linter.rb:138:in `block in visit_children'
/Library/Ruby/Gems/2.0.0/gems/scss-lint-0.37.0/lib/scss_lint/linter.rb:136:in `each'
/Library/Ruby/Gems/2.0.0/gems/scss-lint-0.37.0/lib/scss_lint/linter.rb:136:in `visit_children'
/Library/Ruby/Gems/2.0.0/gems/sass-3.4.13/lib/sass/tree/visitors/base.rb:36:in `block in visit'
/Library/Ruby/Gems/2.0.0/gems/scss-lint-0.37.0/lib/scss_lint/linter/indentation.rb:10:in `visit_root'
/Library/Ruby/Gems/2.0.0/gems/sass-3.4.13/lib/sass/tree/visitors/base.rb:36:in `visit'
/Library/Ruby/Gems/2.0.0/gems/scss-lint-0.37.0/lib/scss_lint/linter.rb:127:in `visit'
/Library/Ruby/Gems/2.0.0/gems/scss-lint-0.37.0/lib/scss_lint/linter.rb:21:in `run'
/Library/Ruby/Gems/2.0.0/gems/scss-lint-0.37.0/lib/scss_lint/runner.rb:52:in `run_linter'
/Library/Ruby/Gems/2.0.0/gems/scss-lint-0.37.0/lib/scss_lint/runner.rb:33:in `block in find_lints'
/Library/Ruby/Gems/2.0.0/gems/scss-lint-0.37.0/lib/scss_lint/runner.rb:31:in `each'
/Library/Ruby/Gems/2.0.0/gems/scss-lint-0.37.0/lib/scss_lint/runner.rb:31:in `find_lints'
/Library/Ruby/Gems/2.0.0/gems/scss-lint-0.37.0/lib/scss_lint/runner.rb:17:in `block in run'
/Library/Ruby/Gems/2.0.0/gems/scss-lint-0.37.0/lib/scss_lint/runner.rb:16:in `each'
/Library/Ruby/Gems/2.0.0/gems/scss-lint-0.37.0/lib/scss_lint/runner.rb:16:in `run'
/Library/Ruby/Gems/2.0.0/gems/scss-lint-0.37.0/lib/scss_lint/cli.rb:54:in `scan_for_lints'
/Library/Ruby/Gems/2.0.0/gems/scss-lint-0.37.0/lib/scss_lint/cli.rb:48:in `act_on_options'
/Library/Ruby/Gems/2.0.0/gems/scss-lint-0.37.0/lib/scss_lint/cli.rb:27:in `run'
/Library/Ruby/Gems/2.0.0/gems/scss-lint-0.37.0/bin/scss-lint:6:in `<top (required)>'
/usr/bin/scss-lint:23:in `load'
/usr/bin/scss-lint:23:in `<main>'
```

**Comment by sds** [comment]
Hey @decodedigital, thanks for the report. 

Unfortunately, I'm having trouble reproducing this issue. From your report, I've tried running the file specified against `scss-lint` 0.37.0 using `sass` 3.4.13 with the version of Ruby that ships with OS X 10.9.4 (`2.0.0p481`), and I don't see any lints reported (more importantly `scss-lint` doesn't crash).

Can I lean on you to implement a fix that solves this problem on your end?

**Comment by kristijanhusak** [comment]
@sds 
I'm having totally same problem as @decodedigital . 

Not sure, but problem may be with the linter not finding `@else` as node for some reason, and returning nil.

**Comment by iamdarrenhall** [comment]
See my comments on issue #409, seems to be the `allow_non_nested_indentation` setting in the Indentation linter.

**Comment by sds** [comment]
Thanks for the additional info, @iamdarrenhall.

Fixed in 28e42c1.
